api/v1alpha1/vcluster_types.go

```diff
@@ -40,7 +40,14 @@ type VClusterSpec struct {
 	// Versions out of the supported range will be ignored, and earliest/latest supported
 	// version will be used instead.
 	// +optional
-	KubernetesVersion *string `json:"kubernetesVersion,omitempty"`
+	KubernetesVersion string `json:"kubernetesVersion,omitempty"`
+
+	VirtualClusterSpec VirtualClusterSpec `json:"virtualClusterSpec,omitempty"`
+}
+
+type VirtualClusterSpec struct {
```

Comment:
Could `Version` and `Image` also be directly inside `VClusterSpec`? IMHO it feels a bit like a duplicate to have a `VirtualClusterSpec` inside the `VClusterSpec`.

Reply:
done